File: code/src/main/java/service/ServicePerformance.java
package service;

import model.Performance;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repository.interfaces.RepositoryInterfacePerformance;
import validator.Validator;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public class ServicePerformance {
    private RepositoryInterfacePerformance repository;
    private Validator<Performance> validator;
    private static final Logger logger = LogManager.getLogger();

    public ServicePerformance(RepositoryInterfacePerformance repository, Validator<Performance> validator) {
        logger.info("Initializing ServicePerformance.");
        this.repository = repository;
        this.validator = validator;
    }

    public Optional<Performance> save(LocalDateTime date, String place, int noOfAvailableSeats, String artist) {
        logger.traceEntry("Saving performance.");
        Performance performance = new Performance(date, place, noOfAvailableSeats, 0, artist);
        validator.validate(performance);
        return repository.save(performance);
    }

    public List<Performance> findAll() {
        logger.info("Finding all the performances.");
        return repository.findAll();
    }

    public List<Performance> findAllForADay(LocalDate date) {
        logger.info("Finding all the performances for the day {}.", date);
        return repository.findAllPerformancesForADay(date);
    }
}

File: MusicFestivalMPP/service/ServicePerformance.cs
using System;
using System.Collections.Generic;
using log4net;
using MusicFestivalMPP.model;
using MusicFestivalMPP.repository.interfaces;
using MusicFestivalMPP.validator;

namespace MusicFestivalMPP.service;

public class ServicePerformance
{
    private IRepositoryPerformance Repository;
    private IValidator<Performance> Validator;
    private static readonly ILog Log = LogManager.GetLogger("ServicePerformance");

    public ServicePerformance(IRepositoryPerformance repository, IValidator<Performance> validator) {
        Log.InfoFormat("Initializing ServicePerformance.");
        Repository = repository;
        Validator = validator;
    }

    public Performance Save(DateTime date, String place, int noOfAvailableSeats, String artist) {
        Log.InfoFormat("Saving performance.");
        Performance performance = new Performance(date, place, noOfAvailableSeats, 0, artist);
        Validator.Validate(performance);
        return Repository.Save(performance);
    }

    public IList<Performance> FindAll() {
        Log.InfoFormat("Finding all the performances.");
        return Repository.FindAll();
    }

    public IList<Performance> FindAllForADay(DateTime date) {
        Log.InfoFormat("Finding all the performances for the day {0}.", date);
        return Repository.FindAllPerformancesForADay(date);
    }
}